### ISSUE TITLE
[helm] `strategy.type=Recreate` not fully supported by all deployments

### ISSUE THREAD

**Issue by jiajie-chen** [apache/airflow#35227]
### Official Helm Chart version

1.11.0 (latest released)

### Apache Airflow version

v2.7.1

### Kubernetes Version

v1.27.4 (minikube v1.31.2)

### Helm Chart configuration

This causes errors:
```yaml
workers:
  strategy:
    type: Recreate
# disable persistence to install `Deployment`
  persistence:
    enabled: false
logs:
  persistence:
    enabled: false
```

But this works:
```yaml
scheduler:
  strategy:
    type: Recreate
# disable persistence to install `Deployment`
logs:
  persistence:
    enabled: false
```

### Docker Image customizations

Using Helm chart defaults.

### What happened

When installing the Helm chart with `workers.strategy.type=Recreate` and persistence disabled, the installation fails.

Helm raises the error:
```
Error: INSTALLATION FAILED: 1 error occurred:
        * Deployment.apps "airflow-worker" is invalid: spec.strategy.rollingUpdate: Forbidden: may not be specified when strategy `type` is 'Recreate'
```

This also applies for `triggerer` and `dagProcessor`.

However, the `scheduler` and `webserver` are able to install with `Recreate`.

### What you think should happen instead

Setting `strategy.type=Recreate` should behave the same for all Airflow `Deployment`s.

The final spec should show something like:
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: airflow-worker
  namespace: airflow
spec:
  strategy:
    type: Recreate
    # no `rollingUpdate`
```

### How to reproduce

Run an installation on a new cluster/namespace (will error):
```bash
helm install airflow apache-airflow/airflow \
  --namespace airflow --create-namespace \
  --set 'workers.strategy.type=Recreate' \
  --set 'workers.persistence.enabled=false' \
  --set 'logs.persistence.enabled=false'
```

However, this doesn't cause errors:
```bash
helm install airflow apache-airflow/airflow \
  --namespace airflow --create-namespace \
  --set 'scheduler.strategy.type=Recreate' \
  --set 'logs.persistence.enabled=false'
```

### Anything else

Setting the strategy to this achieves the same functionality as `Recreate`:
```yaml
workers:
  strategy:
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: "100%"
```

But a consistent fix would be ideal - either not supporting `Recreate` or allowing it for all deployments.

*Edit: using Helm's [default key deletion](https://helm.sh/docs/chart_template_guide/values_files/#deleting-a-default-key) works as well (may not work if using Airflow as a subchart - see below)*

```yaml
workers:
  strategy:
    type: Recreate
    rollingUpdate: null
```

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by hakuno** [comment]
It is because it has a default value right there:

```
  strategy:
    rollingUpdate:
      maxSurge: "100%"
      maxUnavailable: "50%"
```

Your error message says that

> Deployment.apps "airflow-worker" is invalid: spec.strategy.rollingUpdate: Forbidden: may not be specified when strategy `type` is 'Recreate'

Please, can you add this? Or something like that.

```
--set 'workers.strategy.rollingUpdate={}'
```

Thanks in advance.

**Comment by jiajie-chen** [comment]
@hakuno 

Thanks for the advice. I did see that `workers.strategy.rollingUpdate` has a default value set, while `scheduler.strategy` was set to `~`. This led me to try to recreate the issue and submit a ticket.

Trying with `--set 'workers.strategy.rollingUpdate={}'` raised the same `INSTALLATION FAILED` error from Helm.

I also tried `--set 'workers.strategy.rollingUpdate=null'` which *did* work.

However, I think it would be nice to have consistent configuration defaults for `strategy`.
It can be confusing to remember that:
- `workers`/`triggerer`/`dagProcessor` have a default `strategy`, so use Helm's merge semantics to delete `strategy.rollingUpdate` if needed
- `scheduler`/`webserver` has a null default `strategy`
  - `scheduler` by default will omit `strategy` from `Deployment.spec`
  - `webserver` by default will use [a specific `rollingUpdate` value in `Deployment.spec`](https://github.com/apache/airflow/blob/helm-chart/1.11.0/chart/templates/webserver/webserver-deployment.yaml#L53-L72), but doesn't support merging values into this default `rollingUpdate`

**Comment by jiajie-chen** [comment]
I'm also concerned that using `workers.strategy.rollingUpdate=null` may not work if the Airflow chart is used as a subchart by a user.

It looks like Helm has a bug with default key deletion for subcharts:
- Initial bug report: https://github.com/helm/helm/issues/9027
  - Fix for `--set` was merged for Helm v3.13.0: https://github.com/helm/helm/pull/12162
- But it seems there are still edge cases in v3.13:
  - If using `--values`: https://github.com/helm/helm/issues/12488
  - Similar issue, specifically mentions `strategy.type=Recreate`: https://github.com/helm/helm/issues/12469

I will try to create a reproduction of the issue with Airflow as a subchart, when I have time.

**Comment by hakuno** [comment]
@jiajie-chen

> However, I think it would be nice to have consistent configuration defaults for strategy.

Yes, I think so.
